tests/test_event_feedback_displayed.py

```diff
@@ -0,0 +1,165 @@
+
+
+def test_invalid_username_value(valid_event):
+    """ValidationError should be raised if the username value
+    is empty or less than 2 characters
+    """
+    event = valid_event
+    event["username"] = ""
+    with pytest.raises(ValidationError):
+        SCHEMA.load(event)
+    event["username"] = "1"
+    with pytest.raises(ValidationError):
+        SCHEMA.load(event)
+
+
+def test_invalid_event_type_value(valid_event):
+    """ValidationError should be raised if the event_type value
+    is not edx.problem.hint.feedback_displayed
+    """
+    event = valid_event
+    event["event_type"] = "problem_check"
+    with pytest.raises(ValidationError):
+        SCHEMA.load(event)
+
+
+def test_invalid_page_value(valid_event):
+    """ValidationError should be raised if the page value
+    is not x_module
+    """
+    event = valid_event
+    event["page"] = "not_x_module"
+    with pytest.raises(ValidationError):
+        SCHEMA.load(event)
+
+
+def test_invalid_context_org_id_value(valid_event):
+    """ValidationError should be raised if the org_id value
+    is not contained in the course_id
+    """
+    event = valid_event
+    event["org_id"] = "not_org"
+    with pytest.raises(ValidationError):
+        SCHEMA.load(event)
+
+
+def test_invalid_context_path_value(valid_event):
+    """ValidationError should be raised if the path value
+    does not end with problem_check
+    """
+    event = valid_event
+    event["context"]["path"] = "{}_not_problem_check".format(event["context"]["path"])
+    with pytest.raises(ValidationError):
+        SCHEMA.load(event)
+
+
+def test_invalid_event_question_type_value(valid_event):
+    """ValidationError should be raised if the event question_type value
+    is not in the premitted value range
+    """
+    event = valid_event
+    event["event"]["question_type"] = "not_choiceresponse"
+    with pytest.raises(ValidationError):
+        SCHEMA.load(event)
+
+
+def test_invalid_event_with_choice_all_missing(valid_event):
+    """ValidationError should be raised if the event question_type value
+    is choiceresponse and event choice_all is missing
+    """
+    event = valid_event
+    event["event"]["question_type"] = "choiceresponse"
+    with pytest.raises(ValidationError):
+        SCHEMA.load(event)
+    # when the choice_all field is present the event should be valid
+    event["event"]["choice_all"] = ["choice_1"]
+    try:
+        SCHEMA.load(event)
+    except ValidationError:
+        pytest.fail("valid feedback_displayed events should not raise exceptions")
+
+
+def test_invalid_event_with_choice_all_present(valid_event):
+    """ValidationError should be raised if the event choice_all field
+    is present and the event question_type value is not "choiceresponse"
+    """
+    event = valid_event
+    event["event"]["choice_all"] = ["choice_1"]
+    with pytest.raises(ValidationError):
+        SCHEMA.load(event)
```

Comment:
You will need to rewrite those tests when you will have a corresponding factory.